InformationScripting/src/queries/QueryExecutor.cpp

```diff
@@ -57,11 +67,17 @@ Interaction::CommandResult* QueryExecutor::execute()
 		}
 		else
 		{
-			return new Interaction::CommandResult(new Interaction::CommandError(results[0].errors()[0]));
+			hasError = true;
+			errorMessage = results[0].errors()[0];
 		}
 	}
 
-	return new Interaction::CommandResult();
+	if (queries_.empty()) delete this;
```

Comment:
Can't you just post a global event of some sort? We just need anyone to delete it? We could post an event to the scene. There already should be a custom event class for scene, which essentially allows you to post a lambda to be executed.